snuba/web/rpc/common/eap_execute.py

```diff
@@ -0,0 +1,51 @@
+from concurrent.futures import Future, ThreadPoolExecutor
+from typing import Any, Mapping, Optional
+
+from snuba import settings
+from snuba.clickhouse.formatter.query import format_query
+from snuba.datasets.plans.translator.query import identity_translate
+from snuba.datasets.schemas.tables import TableSource
+from snuba.datasets.storages.factory import get_storage
+from snuba.datasets.storages.storage_key import StorageKey
+from snuba.query import Query
+from snuba.query.data_source.simple import Table
+from snuba.reader import Result as ClickhouseQueryResult
+
+
+def eap_execute_query(
+    query: Query,
+    clickhouse_settings: Optional[Mapping[str, Any]] = None,
+    storage_name: str = "eap_spans",
+) -> ClickhouseQueryResult:
+    storage_key = StorageKey(storage_name)
+    storage = get_storage(storage_key)
+    data_source = storage.get_schema().get_data_source()
+    assert isinstance(data_source, TableSource)
+    cluster = storage.get_cluster()
+    clickhouse_query = identity_translate(query)
```

Comment:
@volokluev not sure if this is the best translate, it seems to be used elsewhere in prod but it says it's only for tests?

Reply:
this is fine, feel free to remove that comment too